substrate/runtime/primitives/src/traits.rs

```diff
@@ -102,8 +102,18 @@ impl<T> Convert<T, T> for Identity {
 	fn convert(a: T) -> T { a }
 }
 
+pub trait MaybeEmpty {
+	fn is_empty(&self) -> bool;
+}
+
+impl<T: Default + PartialEq> MaybeEmpty for T {
```

Comment:
this blanket impl is overbroad. The `Default` trait doesn't imply emptiness at all. Are there some types in particular we want this for?

Reply:
yes - specifically for `PublicAux`, for which `AccountId` is used in production environments (but tests use `u64` hence this generalisation). it should eventually be refactored (as i note somewhere near) so that it's actually only implemented for `Option<AccountId>` and `u64`, but for now, it gets the job done.

Reply:
https://github.com/paritytech/polkadot/issues/111